### DISCUSSION TITLE
OpenTheo: Transcribed and searchable Bible teaching with Whisper.cpp

### DISCUSSION THREAD

**Discussion by leiffoged** [ggml-org/whisper.cpp#639] [Show and tell]
Hey all! I just launched [OpenTheo](https://opentheo.com), a free collection of transcribed Bible teachings and other Christian material. It can be found at [https://opentheo.com](https://opentheo.com)

So far, I've transcribed over 10,000 hours of audio/video content for the first time. 

**Transcripts are also fully interactive.** You can click on any words in the transcript to instantly start playing the audio at that point in time. The currently playing portion of the transcript is also underlined to make it easier to follow along. All transcript content is indexed and the site search can be used to find content discussing specific topics or specific sections of the Bible.

Transcripts are created using Whisper.cpp (https://github.com/ggerganov/whisper.cpp). None of this would have been possible without the Whisper model and @ggerganov's work to enable it to run on CPUs. Thank you!!!

<img width="696" alt="Screenshot 2023-03-21 at 5 40 43 PM" src="https://user-images.githubusercontent.com/4405161/226772780-eae772e1-d3bf-4cd3-91a2-13cb4931fddf.png">
<img width="1040" alt="Screenshot 2023-03-21 at 5 41 45 PM" src="https://user-images.githubusercontent.com/4405161/226772878-c780636a-e984-4e08-bfa6-d83ed4ae1da3.png">

**Comment by jmantn** [comment]
Nice work!

I seen your post and it got me wondering if I could have html files on my computer and the audio files in the same folder for example and clicked on text generated by whisper and have it jump to that position in the audio.  

I haven’t done much coding in 20 years but was wondering if that was possible. I know with whisper it can output timestamps but unsure and if the functionality is done with html and or JavaScript or something else.  

Thanks for any help or pointers.

**Reply by pcbua** [reply]
It looks impressive.
Can you share the model name and parameters you used?
Your recordings are pretty long, I didn't spot hallucinations in your results. But I see them in mine.
Was pre- or post-processing applied to your audio or transcription?

**Reply by leiffoged** [reply]
@jmantn Yeah, you'd have to write some JavaScript to hook up the HTML for the transcript with the audio player.

@pcbua Default parameters. Some recordings use base.en model, others use small.en model. Cost of medium/large models isn't worth it for most recordings IMO. Metadata about the recording (title, author, album name, description) are fed in to the 'prompt' in complete sentences (e.g. "The title of this recording is X. The author of this recording is Y....").

No pre/post-processing of the audio/text, aside from formatting the transcript output into more readable paragraphs.
